lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -982,6 +987,265 @@ def _generate_report(self, trans, invocation_id, **kwd):
+                            }
+                            output_list.append(output_obj)
+
+                        if job_output.name in workflow_outputs_list:
+                            output = {
+                                'mediatype': job_output.dataset.extension,
+                                'uri': {
+                                    'filename': job_output.dataset.name,
+                                    'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
+                                    'access_time': job_output.dataset.create_time.isoformat(),
+                                }
+                            }
+                            output_subdomain.append(output)
+                current_tool = dict_workflow['steps'][str(i)]
+                pipeline_step = {
+                    'step_number': i,
+                    'name': current_tool['name'],
+                    'description': current_tool['annotation'],
+                    'version': current_tool['tool_version'],
+                    'prerequisite': kwd.get('prerequisite', []),
+                    'input_list': input_list,
+                    'output_list': output_list
+                }
+                pipeline_steps.append(pipeline_step)
+                try:
+                    software_prerequisite = {
+                        'name': current_tool['tool_shed_repository']['name'],
+                        'version': current_tool['tool_version'],
+                        'uri': {
+                            'uri': current_tool['content_id'],
+                            'access_time': current_tool['uuid']
+                        }
+                    }
+                    if software_prerequisite['uri']['uri'] not in tools:
+                        software_prerequisites.append(software_prerequisite)
+                        tools.append(software_prerequisite['uri']['uri'])
+                except Exception:
+                    continue
+
+            if step.workflow_step.type == 'data_input':
+                for output_assoc in step.output_datasets:
+                    encoded_dataset_id = trans.security.encode_id(output_assoc.dataset.id)
+                input_obj = {
+                    'filename': step.workflow_step.label,
+                    'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
+                    'access_time': step.workflow_step.update_time.isoformat(),
+                }
+                input_subdomain.append(input_obj)
+
+            if step.workflow_step.type == 'data_collection_input':
+                encoded_dataset_id = trans.security.encode_id(step.workflow_step.id)
+                input_obj = {
+                    'filename': step.workflow_step.label,
+                    'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
```

Comment:
I don't think this is correct? Does it work correctly?

Reply:
I think you are correct. That must be a mistake.

Reply:
Would this work instead?
```
                for output_assoc in step.output_datasets:
                    encoded_dataset_id = trans.security.encode_id(output_assoc.dataset_collection.id)
                    input_obj = {
                        'filename': step.workflow_step.label,
                        'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
                        'access_time': step.workflow_step.update_time.isoformat(),
                    }
                    input_subdomain.append(input_obj)
```